src/sources/geniuscom.cpp

```diff
@@ -71,52 +74,119 @@ std::vector<LyricDataRaw> GeniusComSource::search(const LyricSearchParams& param
         return {};
     }
 
-    LOG_INFO("Page %s retrieved", url.c_str());
-    std::string lyric_text;
-    pugi::xml_document doc;
-    load_html_document(content.c_str(), doc);
+    std::vector<LyricDataRaw> lyrics;
 
-    const char* xpath_queries[] = { "//div[@class='lyrics']", "//div[contains(@class, 'Lyrics__Container')]" };
-    for(const char* query_str : xpath_queries)
-    {
-        pugi::xpath_query query_lyricdivs(query_str);
-        pugi::xpath_node_set lyricdivs = query_lyricdivs.evaluate_node_set(doc);
+    {   // Parser gets its own scope
+        const cJSON* searchResponse = NULL;
+        const cJSON* searchHits = NULL;
+        const cJSON* searchHit = NULL;
+        const cJSON* searchResult = NULL;
+        const cJSON* searchPath = NULL;
 
-        if(!lyricdivs.empty())
-        {
-            for(const pugi::xpath_node& node : lyricdivs)
+        cJSON* searchRes = cJSON_Parse(content.c_str());
+        if (searchRes == NULL) {

```

Comment:
We should in general prefer `nullptr` over `NULL`:

```suggestion
        if (searchRes == nullptr) {
```